qualification/recv.py

```diff
@@ -157,12 +161,8 @@ async def complete_chunks(
                     timestamp = chunk.chunk_id * self.timestamp_step
                     if min_timestamp is not None and timestamp < min_timestamp:
                         logger.debug("Skipping chunk with timestamp %d (< %d)", timestamp, min_timestamp)
-                    elif not np.all(chunk.present):
+                    elif not self.is_complete_chunk(chunk):
                         logger.debug("Incomplete chunk %d", chunk.chunk_id)
-                    elif (chunk.data.dtype == np.dtype(np.int32) and np.any(chunk.data == -(2**31))) or (
-                        chunk.extra is not None and np.min(chunk.extra) < self.n_ants
```

Comment:
I know this was a slightly different check to that now on Line 358, *but*
- I see this had a `np.min(chunk.extra) < self.n_ants`, and
- Now it's `== self.n_ants`

Was the check not entirely correct to begin with? Or has something changed in how we(/you) make use of `chunk.extra` now?

Reply:
It was correct in the original code. But the original code is a check for an *incomplete* chunk. The new code is a check for a *complete* chunk, so the test needed to be inverted. Technically inverting it would be `>= self.n_ants` but it can't be more than `n_ants` (and if it is, we probably want to reject it anyway).

Reply:
Noted and noted.